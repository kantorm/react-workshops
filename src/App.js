import React, { Component } from 'react';
import {connect} from "react-redux";
import logo from './logo.svg';
import './App.css';
import {addTodo, removeTodo} from "./store";


export class TodoAdd extends Component {
  constructor(props) {
    super(props);

    this.state = {
      value: "",
    };
  }

  updateValue = ({target: {value}}) => {
    this.setState({value})
  }

  addTodo = () => {
    this.props.addTodo(this.state.value);
    this.setState({value: ""});
  }

  render() {
    return (
      <div>
        <input onChange={this.updateValue} value={this.state.value}/>
        <button onClick={this.addTodo}>Add todo</button>
      </div>
    );
  }
}

export class TodoList extends Component {
  render() {
    const {todos} = this.props;
    return (
      <div>
        {todos.map((todo) => {
          return (
            <li key={todo.id}>
              {todo.text}
              <button onClick={() => this.props.removeTodo(todo.id)}>REMOVE</button>
            </li>
          )
        })}
      </div>
    );
  }
}

class App extends Component {
  addTodo = (todo) => this.props.addTodo(todo);

  removeTodo = (todoId) => this.props.removeTodo(todoId);

  render() {
    return (
      <div className="App">
        <div className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
        </div>
        <TodoAdd addTodo={this.addTodo}/>
        <TodoList todos={this.props.todos} removeTodo={this.removeTodo}/>
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  todos: state.todos,
});

const mapDispatchToProps = {
  addTodo,
  removeTodo,
}

export default connect(mapStateToProps, mapDispatchToProps)(App);
