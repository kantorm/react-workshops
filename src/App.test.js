import React from 'react';
import { shallow } from "enzyme";
import ReactDOM from 'react-dom';
import App, { TodoList, TodoAdd } from './App';


it('renders without crashing', () => {
  const wrapper = shallow(<App />);
  expect(wrapper).toMatchSnapshot();
});

it("renders TodoList", () => {
  const wrapper = shallow(<TodoList />);
  expect(wrapper).toMatchSnapshot();
});
