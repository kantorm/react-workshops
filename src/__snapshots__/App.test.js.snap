// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="App"
>
    <div
        className="App-header"
    >
        <img
            alt="logo"
            className="App-logo"
            src="logo.svg"
        />
    </div>
    <TodoAdd
        addTodo={[Function]}
    />
    <TodoList
        removeTodo={[Function]}
        todos={
            Array [
                Object {
                  "id": 1,
                  "text": "todo 1",
                },
                Object {
                  "id": 2,
                  "text": "todo 2",
                },
              ]
        }
    />
</div>,
  "nodes": Array [
    <div
      className="App"
>
      <div
            className="App-header"
      >
            <img
                  alt="logo"
                  className="App-logo"
                  src="logo.svg"
            />
      </div>
      <TodoAdd
            addTodo={[Function]}
      />
      <TodoList
            removeTodo={[Function]}
            todos={
                  Array [
                        Object {
                          "id": 1,
                          "text": "todo 1",
                        },
                        Object {
                          "id": 2,
                          "text": "todo 2",
                        },
                      ]
            }
      />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <App />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": App {
        "_reactInternalInstance": [Circular],
        "addTodo": [Function],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "removeTodo": [Function],
        "state": Object {
          "todos": Array [
            Object {
              "id": 1,
              "text": "todo 1",
            },
            Object {
              "id": 2,
              "text": "todo 2",
            },
          ],
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="App"
>
          <div
                    className="App-header"
          >
                    <img
                              alt="logo"
                              className="App-logo"
                              src="logo.svg"
                    />
          </div>
          <TodoAdd
                    addTodo={[Function]}
          />
          <TodoList
                    removeTodo={[Function]}
                    todos={
                              Array [
                                        Object {
                                          "id": 1,
                                          "text": "todo 1",
                                        },
                                        Object {
                                          "id": 2,
                                          "text": "todo 2",
                                        },
                                      ]
                    }
          />
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="App"
>
          <div
                    className="App-header"
          >
                    <img
                              alt="logo"
                              className="App-logo"
                              src="logo.svg"
                    />
          </div>
          <TodoAdd
                    addTodo={[Function]}
          />
          <TodoList
                    removeTodo={[Function]}
                    todos={
                              Array [
                                        Object {
                                          "id": 1,
                                          "text": "todo 1",
                                        },
                                        Object {
                                          "id": 2,
                                          "text": "todo 2",
                                        },
                                      ]
                    }
          />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <App />,
}
`;
